src/SfxWeb/src/app/views/cluster/federation/federation.component.ts

```diff
@@ -0,0 +1,656 @@
+import { Component, Injector, OnDestroy, OnInit } from '@angular/core';
+import { Observable, of, Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+import { IResponseMessageHandler } from 'src/app/Common/ResponseMessageHandlers';
+import { NodeEventList, PartitionEventList } from 'src/app/Models/DataModels/collections/Collections';
+import { NodeEvent, PartitionEvent } from 'src/app/Models/eventstore/Events';
+import { IOnDateChange } from 'src/app/modules/event-store/double-slider/double-slider.component';
+import { IEventStoreData, IQuickDates } from 'src/app/modules/event-store/event-store/event-store.component';
+import { IOptionConfig } from 'src/app/modules/event-store/option-picker/option-picker.component';
+import { DataService } from 'src/app/services/data.service';
+import { TelemetryService } from 'src/app/services/telemetry.service';
+import { TimeUtils } from 'src/app/Utils/TimeUtils';
+import { BaseControllerDirective } from 'src/app/ViewModels/BaseController';
+import { Network} from "vis-network/standalone"
+
+// structure that has all the relevant info about particular node
+export interface NodeData {
```

Comment:
actually, all of them have to be present all the time, because we're using two maps.
I've added question marks "just in case" but totally unnecessary. thank you for noticing :)